tests/Common/FilesTest.php

```diff
@@ -228,45 +235,42 @@ public function testFileUploadUsingFederationToken($encrypted)
 
         $result = $this->_client->prepareFileUpload($options);
 
-        $uploadParams = $result['uploadParams'];
-        $this->assertArrayHasKey('credentials', $uploadParams);
-
-        $s3Client = new \Aws\S3\S3Client([
-            'version' => 'latest',
-            'region' => $result['region'],
-            'credentials' => [
-                'key' => $uploadParams['credentials']['AccessKeyId'],
-                'secret' => $uploadParams['credentials']['SecretAccessKey'],
-                'token' => $uploadParams['credentials']['SessionToken'],
-            ],
-        ]);
-
-        $putParams = array(
-            'Bucket' => $uploadParams['bucket'],
-            'Key' => $uploadParams['key'],
-            'Body' => fopen($pathToFile, 'r+'),
-        );
-
-        if ($options->getIsEncrypted()) {
-            $putParams['ServerSideEncryption'] = $uploadParams['x-amz-server-side-encryption'];
+        if ($result['provider'] === 'azure') {
+            $uploadParams = $result['absUploadParams'];
+            $this->assertArrayHasKey('absCredentials', $uploadParams);
+        } else {
+            $uploadParams = $result['uploadParams'];
+            $this->assertArrayHasKey('credentials', $uploadParams);
         }
 
-        $s3Client->putObject($putParams);
+        $fileId = $this->_client->uploadFile($pathToFile, $options);
 
-        $file = $this->_client->getFile($result['id']);
+        $file = $this->_client->getFile($fileId);
 
         $this->assertEquals(file_get_contents($pathToFile), file_get_contents($file['url']));
-        $this->assertEquals($result['isEncrypted'], $options->getIsEncrypted());
 
-        try {
-            $s3Client->putObject(array(
-                'Bucket' => $uploadParams['bucket'],
-                'Key' => $uploadParams['key'] . '_part0001',
-                'Body' => fopen($pathToFile, 'r+'),
-            ));
-            $this->fail('Access denied exception should be thrown');
-        } catch (\Aws\S3\Exception\S3Exception $e) {
-            $this->assertEquals(403, $e->getStatusCode());
+        if ($result['provider'] === 'aws') {
```

Comment:
Z tohodle testu zbylo takový divný torzo. Původně testoval zápis non-sliced souboru fedration tokenem a to že nemá právo zapsat do jiné cesty.
Asi bych ho udělal separátní pro Aws a Azure tzn. založit pro takovéhle testy FilesAzureTest.php a FilesAwsTest.php. Dává ti to smyl?

Reply:
Máme to podobně pro tabulkové backendy, kompatibilní features jsou pokryté společnými testy. a pro backend specifické featury jsou pak samotstné testy https://github.com/keboola/storage-api-php-client/tree/master/tests/Backend/Snowflake

Reply:
Jo. Udělám CommonTest, Aws a Azure. Common přidám do `common` testsuity a pro AWS a Azure bych udělal nový testsuite. Dává ti smysl udělat pro tyhle dva testsuity 2 nový projekty ? Nebo bys AWS hodil do common? (předpokládám, že common = SNFLK + S3) ?

Reply:
asi bych hodil aws do common. Takže přidat zatím azure testsuite. ten common neřeší vůbec backend takže tam na SNFLK/RS nezáleží.

Reply:
Uvidíme jak to bude vypadat kdyžtak se to pak ještě poladí. Teď by se tada založil jeden projekt navíc který bude mít ABS registrovaný jako file storage.

Reply:
On nejde vytvořit projekt, který bude na ABS. Teď jsem to zkusil a padlo to. Myslím, že to je tím, že se logy pořád pevně ukládají na S3. Můžu to jít vymyslet, ale asi bych spíš dodělal ten load/unload snflk a pak dodělal tohle. Takhle bych stejně mohl spouštět jenom ten jeden test, ale po load/unload, už by toho mělo fungovat mnohem víc (možná všechno) a budu tam moc spustit třeba i snowflake testy.

Reply:
Ještě se na to podívám, protože u sebe projekt assignutý na ABS mám. Takže by to mělo jít.

Reply:
Jo tak už to frčí. Jdu to donastavit.

Reply:
Přidal jsem to do buildu travisu https://github.com/keboola/connection/pull/2194/commits/f2a4e6150ce1757f1323e7b13398f7f44843bb16.

Udělal bych to tak, že tohle nasadím do masteru a pak otaguju branch a spustím testy. U tohohle není problém, když to bude v masteru.

Reply:
logy s tím vůbec nesouvisejí ty se budou řešit pak v další fázi